### PR TITLE
New attribute type 'weakness'

### PR DESCRIPTION
**Pull request by chrisr3d**
New attribute type to describe a common weakness enumeration with its id and link in event view to its description in [CIRCL CVE Search](https://cve.circl.lu/cwe)

### CODE REVIEW

**Review by adulau — APPROVED** [review]
Good to me. I also prefer weakness as generic name like we do for vulnerability if new standards appear in the future. Thanks for the contribution.

File: app/View/Elements/Events/View/value_field.ctp [hunk 1]
```diff
@@ -35,6 +35,9 @@
     } else if ('vulnerability' == $object['type']) {
         $cveUrl = (is_null(Configure::read('MISP.cveurl'))) ? "http://www.google.com/search?q=" : Configure::read('MISP.cveurl');
         echo $this->Html->link($sigDisplay, $cveUrl . $sigDisplay, array('target' => '_blank', 'class' => $linkClass));
+    } else if ('weakness' == $object['type']) {
+        $cweUrl = (is_null(Configure::read('MISP.cweurl'))) ? "http://www.google.com/search?q=" : Configure::read('MISP.cweurl');
+        echo $this->Html->link($sigDisplay, $cweUrl . explode("-", $sigDisplay)[1], array('target' => '_blank', 'class' => $linkClass));
```
Comment (iglocska) [inline app/View/Elements/Events/View/value_field.ctp:40]: Default to 'http://cve.circl.lu/cwe/'

File: app/View/Elements/Feeds/View/value_field.ctp [hunk 2]
```diff
@@ -26,6 +26,9 @@
   } else if ('vulnerability' == $object['type']) {
     $cveUrl = (is_null(Configure::read('MISP.cveurl'))) ? "http://www.google.com/search?q=" : Configure::read('MISP.cveurl');
     echo $this->Html->link($sigDisplay, $cveUrl . $sigDisplay, array('target' => '_blank', 'class' => $linkClass));
+  } else if ('weakness' == $object['type']) {
+    $cweUrl = (is_null(Configure::read('MISP.cweurl'))) ? "http://www.google.com/search?q=" : Configure::read('MISP.cweurl');
```
Comment (iglocska) [inline app/View/Elements/Feeds/View/value_field.ctp]: 'http://cve.circl.lu/cwe/'

File: app/View/Elements/Servers/View/value_field.ctp [hunk 3]
```diff
@@ -26,6 +26,9 @@
   } else if ('vulnerability' == $object['type']) {
     $cveUrl = (is_null(Configure::read('MISP.cveurl'))) ? "http://www.google.com/search?q=" : Configure::read('MISP.cveurl');
     echo $this->Html->link($sigDisplay, $cveUrl . $sigDisplay, array('target' => '_blank', 'class' => $linkClass));
+  } else if ('weakness' == $object['type']) {
+    $cweUrl = (is_null(Configure::read('MISP.cweurl'))) ? "http://www.google.com/search?q=" : Configure::read('MISP.cweurl');
```
Comment (iglocska) [inline app/View/Elements/Servers/View/value_field.ctp]: 'http://cve.circl.lu/cwe/'

File: app/Model/Server.php [hunk 4]
```diff
@@ -530,6 +530,14 @@ public function __construct($id = false, $table = null, $ds = null)
                                 'test' => 'testForEmpty',
                                 'type' => 'string',
                         ),
+                        'cweurl' => array(
+                                'level' => 1,
+                                'description' => __('Turn Weakness type attributes into links linking to the provided CWE lookup'),
+                                'value' => '',
```
Comment (iglocska) [inline app/Model/Server.php]: as value, put the URL that the user should expect. This is not acted on, it's display only to describe the default behaviour.

'http://cve.circl.lu/cwe/'

**Comment by iglocska** [comment]
Thanks a lot kind sir!

**Pull request merged**